presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PlanNodeDecorrelator.java

```diff
@@ -277,29 +278,52 @@ public PlanNodeDecorrelator(PlanNodeIdAllocator idAllocator, PlanVariableAllocat
 
         private Multimap<VariableReferenceExpression, VariableReferenceExpression> extractCorrelatedSymbolsMapping(List<Expression> correlatedConjuncts)
         {
-            // TODO: handle coercions and non-direct column references
```

Comment:
I'm not sure what non-direct column references is specifically, but I'm suspecting it's if you had some kind of operation on the column, e.g t1.a + 1 = t2.a instead of just t1.a = t2.a. I guess that is still unsupported.

Reply:
yes I was planning to do it separatelly. Non-direct column references like you point out may have different expression type like ArithmeticBinaryExpression, or `t1.a = -t2.a` for ArithmeticUnaryExpression, or `t1.a != t2.a` for NotExpression. For now I think euqality expression is the most common use case so I am adding this feature within the commit. I will add this comment as a TODO later

Reply:
Marking that with the latest implementation, we won't have any other non-directive column reference decorrelation issues